dev/com.ibm.ws.threading/src/com/ibm/ws/threading/internal/ThreadPoolController.java

```diff
@@ -1558,7 +1573,7 @@ private boolean resolveHang(long tasksCompleted, boolean queueEmpty, int poolSiz
                 }
             }
         }
-        return actionTaken;
+        return hangDetected;
```

Comment:
This really changes the meaning of this method as well.  Now whether we make a change or not, we say we resolved the hang to the caller and it doesn't continue with the algorithm.  Is that what you want?